### PR TITLE
feat(server): REST endpoint for secrets management

### LINKED ISSUE #11989: REST API: Secrets management endpoint
Parent: #11985

## Summary

Secrets store encrypted LLM provider credentials (API keys for OpenAI, Anthropic, Azure, etc.) in the Phoenix database. They are the mechanism by which Phoenix resolves credentials for evaluators, playground chat, and any LLM-backed operation. Today there is no REST endpoint for managing secrets — all management must go through GraphQL or the UI.

This blocks programmatic environment setup: provisioning a Phoenix instance with provider credentials requires either GraphQL knowledge or manual UI interaction.

### Data Model

| Field | Type | Notes |
|---|---|---|
| `key` | str (PK) | ASCII-only, the secret name (e.g., `OPENAI_API_KEY`) |
| `value` | bytes | Fernet-encrypted (AES-128-CBC + HMAC-SHA256) |
| `user_id` | int? (FK → users) | SET NULL on delete — tracks who last modified |
| `updated_at` | datetime | Auto-updated on insert/update |

### Encryption

- Uses Fernet encryption derived from `PHOENIX_SECRET` environment variable via PBKDF2-HMAC-SHA256 (600k iterations).
- Values are encrypted before DB write — a SQLAlchemy event listener enforces this as a defense-in-depth measure.
- Decryption failures return an `UnparsableSecret` type rather than crashing.

### How Secrets Are Used

Secrets are resolved in this priority order for LLM operations:
1. Explicitly provided credentials (per-request)
2. **Database secrets** (this system)
3. Environment variables

Common secret keys: `OPENAI_API_KEY`, `ANTHROPIC_API_KEY`, `AZURE_OPENAI_API_KEY`, plus provider-specific keys for Google, Bedrock, Vertex AI, etc.

### Existing GraphQL Operations

| Operation | Permissions |
|---|---|
| `upsertOrDeleteSecrets(secrets: [{key, value?}])` | ADMIN (when auth enabled) |
| `secrets(keys?: [str])` query | All authenticated (values: ADMIN only) |

The mutation is atomic: each entry with a `value` is upserted; entries with `value: null` are deleted. Duplicate keys in the same request are deduplicated (last wins).

## Proposed Endpoints

> Required role: **ADMIN** (when auth is enabled)

- [ ] [rest] `PUT /secrets` — upsert or delete secrets (atomic batch operation)

### Request Shape

```json
{
  "secrets": [
    { "key": "OPENAI_API_KEY", "value": "sk-..." },
    { "key": "ANTHROPIC_API_KEY", "value": "sk-ant-..." },
    { "key": "OLD_UNUSED_KEY", "value": null }
  ]
}
```

- `value: "<string>"` → upsert (create or update)
- `value: null` → delete
- Duplicate keys: last occurrence wins

### Response Shape (200)

```json
{
  "data": {
    "upserted_keys": ["OPENAI_API_KEY", "ANTHROPIC_API_KEY"],
    "deleted_keys": ["OLD_UNUSED_KEY"]
  }
}
```

## Use Cases

1. **Environment provisioning**: Script sets up a new Phoenix instance with provider API keys in one call
2. **Key rotation**: Update provider keys across environments programmatically
3. **Infrastructure-as-code**: Manage LLM provider credentials alongside other infra config
4. **Cleanup**: Remove stale provider keys when decommissioning integrations

## Implementation Notes

- Use `Depends(require_admin)` and `Depends(is_not_locked)`.
- Encryption/decryption functions are available on `request.app.state` context (the `encrypt` and `decrypt` callables from the Fernet key).
- Input validation: keys must be non-empty ASCII (after trimming), values must be non-empty strings (after trimming) or `null`.
- The batch must contain at least one entry.
- Idempotent deletes: deleting a non-existent key succeeds silently.
- Secret values should **never** be returned in the response — only confirm which keys were upserted/deleted.

## Key Files

- DB model: `src/phoenix/db/models.py` (Secret, lines ~2471-2497)
- GraphQL mutation: `src/phoenix/server/api/mutations/secret_mutations.py`
- Encryption: `src/phoenix/server/encryption.py`
- Credential resolution: `src/phoenix/server/api/helpers/playground_clients.py`
- GraphQL query: `src/phoenix/server/api/queries.py` (secrets, lines ~274-304)

### PR DESCRIPTION
**Pull request by mikeldking**
## Summary

Closes #11989

Adds `PUT /v1/secrets` — a batch upsert/delete endpoint for managing encrypted LLM provider credentials programmatically.

- **Admin-only**: guarded by `require_admin` + `is_not_locked` dependencies
- **Upsert**: `value: "<string>"` creates or updates the secret (Fernet-encrypted before DB write)
- **Delete**: `value: null` removes the secret; deleting a non-existent key succeeds silently
- **Deduplication**: when the same key appears multiple times, the last occurrence wins
- **Audit**: `user_id` is recorded on upsert (consistent with GraphQL mutation behavior)
- **No secret exposure**: values are never returned in any response

### Request

```http
PUT /v1/secrets
Content-Type: application/json

{
  "secrets": [
    { "key": "OPENAI_API_KEY",    "value": "sk-..." },
    { "key": "ANTHROPIC_API_KEY", "value": "sk-ant-..." },
    { "key": "OLD_UNUSED_KEY",    "value": null }
  ]
}
```

### Response

```json
{
  "data": {
    "upserted_keys": ["OPENAI_API_KEY", "ANTHROPIC_API_KEY"],
    "deleted_keys": ["OLD_UNUSED_KEY"]
  }
}
```

## Testing

Tests moved from unit to integration following the project's integration test pattern:

- **30 integration tests** in `tests/integration/secrets/` across 5 classes:
  - `TestSecretsEncryptionRoundtrip` — E2E verification: write via REST → read via GraphQL `secrets` query → assert decrypted value matches original plaintext
  - `TestSecretsCRUDViaREST` — CRUD operations through `PUT /v1/secrets`
  - `TestSecretsCRUDViaGraphQL` — CRUD through GraphQL mutation
  - `TestSecretsAuthorization` — admin-only access for REST and GraphQL (admin, member, viewer, unauthenticated)
  - `TestSecretsValidation` — validation edge cases through the real HTTP stack
- **7 unit tests** retained in `tests/unit/` for Pydantic validation and OpenAPI schema assertions

## Files Changed

| File | Change |
|------|--------|
| `src/phoenix/server/api/routers/v1/secrets.py` | New router with `PUT /secrets` endpoint |
| `src/phoenix/server/api/routers/v1/__init__.py` | Register `secrets_router` |
| `tests/integration/secrets/conftest.py` | Integration test fixtures (server with auth) |
| `tests/integration/secrets/test_secrets.py` | 30 integration tests |
| `tests/unit/server/api/routers/v1/test_secrets.py` | Trimmed to validation/schema tests only |
| `tests/integration/_helpers.py` | Add endpoint to `_ADMIN_ONLY_ENDPOINTS` |

## How to Verify

1. Run integration tests: `pytest tests/integration/secrets/ -v`
2. Run remaining unit tests: `pytest tests/unit/server/api/routers/v1/test_secrets.py -v`
3. Start Phoenix with auth enabled, confirm `PUT /v1/secrets` works with admin token and is rejected for non-admin roles

### CODE REVIEW

File: src/phoenix/server/api/routers/v1/secrets.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    @field_validator("key")
+    @classmethod
+    def validate_key(cls, v: str) -> str:
+        """Keys must be non-empty ASCII strings (after trimming)."""
+        v = v.strip()
+        if not v:
+            raise ValueError("Key cannot be empty")
+        if not v.isascii():
+            raise ValueError("Key must be ASCII")
```
Comment (RogerHYang) [inline src/phoenix/server/api/routers/v1/secrets.py]: probably can be more strict to start: i.e. no punctuations, no white spaces, etc.

File: tests/integration/client/test_secrets.py
Comment (RogerHYang) [inline tests/integration/client/test_secrets.py:1]: FWIW, we've been putting REST API tests in `tests/integration/client/`. This lets us efficiently test both `phoenix-client` and the server in one pass. For endpoints without client support (e.g. v1/users), we keep them there too — they share the same server fixture, and it's easier to find everything in one place.

File: requirements/ci.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 mypy==1.19.1
+numpy<2.3; python_version < "3.12"
```
Comment (mikeldking) [inline requirements/ci.txt:2]: avoids randbits issue:

seems integration tests are failing:

integration_tests: commands_pre[2] /home/runner/work/phoenix/phoenix/tests/integration> python -c 'import phoenix.version as v; major,*_=v.__version__.split('"'"'.'"'"'); nv=f'"'"'{int(major)+1}.0.0'"'"'; open(v.__file__,'"'"'w'"'"').write(f'"'"'__version__ = "{nv}"\n'"'"')'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/phoenix/__init__.py", line 13, in <module>
    from .session.client import Client
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/phoenix/session/client.py", line 14, in <module>
    import pandas as pd
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/pandas/__init__.py", line 37, in <module>
    from pandas._config import (
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/pandas/_config/__init__.py", line 20, in <module>
    from pandas._config import config
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/pandas/_config/config.py", line 68, in <module>
    from pandas._typing import (
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/pandas/_typing.py", line 198, in <module>
    np.random.Generator,
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/numpy/__init__.py", line 340, in __getattr__
    import numpy.random as random
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/numpy/random/__init__.py", line 180, in <module>
    from . import _pickle
  File "/home/runner/work/phoenix/phoenix/.tox/integration_tests/lib/python3.10/site-packages/numpy/random/_pickle.py", line 1, in <module>
    from .bit_generator import BitGenerator
  File "numpy/random/bit_generator.pyx", line 40, in init numpy.random.bit_generator
ImportError: cannot import name randbits
integration_tests: exit 1 (0.68 seconds) /home/runner/work/phoenix/phoenix/tests/integration> python -c 'import phoenix.version as v; major,*_=v.__version__.split('"'"'.'"'"'); nv=f'"'"'{int(major)+1}.0.0'"'"'; open(v.__file__,'"'"'w'"'"').write(f'"'"'__version__ = "{nv}"\n'"'"')' pid=2939
  integration_tests: FAIL code 1 (18.25=setup[1.38]+cmd[14.88,1.31,0.68] seconds)
  evaluation failed :( (18.29 seconds)

**Review by RogerHYang — APPROVED** [review]

File: requirements/ci.txt (hunk 2 above)
Comment (RogerHYang) [reply requirements/ci.txt:2]: first time seeing this
